src/codecoverage/bot/code_coverage_bot/phabricator.py

```diff
@@ -123,6 +122,18 @@ def generate(self, report):
     def upload(self, report):
         results = self.generate(report)
 
-        for revision_id, coverage in results.items():
-            # TODO: Actually upload coverage data to Phabricator.
-            logger.info('{} coverage: {}'.format(revision_id, coverage))
+        if secrets[secrets.PHABRICATOR_ENABLED]:
+            phabricator = PhabricatorAPI(secrets[secrets.PHABRICATOR_TOKEN], secrets[secrets.PHABRICATOR_URL])
+        else:
+            phabricator = None
+
+        for rev_id, coverage in results.items():
+            logger.info('{} coverage: {}'.format(rev_id, coverage))
+
+            if not phabricator or not coverage:
+                continue
+
+            rev_data = phabricator.load_revision(rev_id=rev_id)
+            phabricator.upload_coverage_results(rev_data['fields']['diffPHID'], coverage)
+            # XXX: This is only necessary until https://bugzilla.mozilla.org/show_bug.cgi?id=1487843 is resolved.
+            phabricator.upload_lint_results(rev_data['fields']['diffPHID'], 'pass', [])
```

Comment:
So you already had the implementation but were not using it ?

Reply:
Yes, for a few months :smile: